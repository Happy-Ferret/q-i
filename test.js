const qi = require('./index');

/* eslint-disable id-blacklist, no-console */

describe('stringify()', () => {
	it('should color highlight basic types', () => {
		const result = qi.stringify({
			a: undefined,
			b: null,
			c: 0,
			d: 42,
			e: '42',
			f: false,
			g: true,
			h: /pizza/,
			i: [1, 2],
			j: { x: 1, y: 2 },
		});
		expect(result).toMatchSnapshot();
	});

	it('should print contructor names for classes', () => {
		class Pizza {
			constructor({ name }) {
				this.name = name;
			}
		}
		const result = qi.stringify({
			x: new Pizza({ name: 'Quattro Formaggi' }),
		});
		expect(result).toMatchSnapshot();
	});

	it('should collapse arrays with more than 30 items', () => {
		const result = qi.stringify({
			a: [1, 2],
			b: [
				1,
				2,
				3,
				4,
				5,
				6,
				7,
				8,
				9,
				10,
				11,
				12,
				13,
				14,
				15,
				16,
				17,
				18,
				19,
				20,
				21,
				22,
				23,
				24,
				25,
				26,
				27,
				28,
				29,
				30,
				31,
			],
		});
		expect(result).toMatchSnapshot();
	});

	it('should collapse objects with more than 30 keys', () => {
		const result = qi.stringify({
			a: { k1: 0, k2: 0 },
			b: {
				k1: 0,
				k2: 0,
				k3: 0,
				k4: 0,
				k5: 0,
				k6: 0,
				k7: 0,
				k8: 0,
				k9: 0,
				k10: 0,
				k11: 0,
				k12: 0,
				k13: 0,
				k14: 0,
				k15: 0,
				k16: 0,
				k17: 0,
				k18: 0,
				k19: 0,
				k20: 0,
				k21: 0,
				k22: 0,
				k23: 0,
				k24: 0,
				k25: 0,
				k26: 0,
				k27: 0,
				k28: 0,
				k29: 0,
				k30: 0,
				k31: 0,
			},
		});
		expect(result).toMatchSnapshot();
	});
});

describe('print()', () => {
	const console$log = console.log;
	beforeEach(() => {
		console.log = jest.fn();
	});
	afterEach(() => {
		console.log = console$log;
	});

	it('should print an object', () => {
		qi.print({
			a: 42,
		});
		expect(console.log).toBeCalledWith(expect.stringMatching('42'));
	});
});
