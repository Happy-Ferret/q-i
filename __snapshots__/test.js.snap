// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stringify() should accept address and custom max items 1`] = `
"{
  c: [2mObject {4}[22m
}"
`;

exports[`stringify() should accept custom max items 1`] = `
"{
  a: {
    k1: [36m0[39m,
    k2: [36m0[39m
  },
  b: [2mObject {4}[22m
}"
`;

exports[`stringify() should collapse arrays with more than 30 items 1`] = `
"{
  a: [
    [36m1[39m,
    [36m2[39m
  ],
  b: b,[2mArray[31][22m
}"
`;

exports[`stringify() should collapse objects with more than 30 keys 1`] = `
"{
  a: {
    k1: [36m0[39m,
    k2: [36m0[39m
  },
  b: [2mObject {31}[22m
}"
`;

exports[`stringify() should color highlight basic types 1`] = `
"{
  a: [3mundefined[23m,
  b: [3mnull[23m,
  c: [36m0[39m,
  d: [36m42[39m,
  e: [32m'42'[39m,
  f: [35mfalse[39m,
  g: [35mtrue[39m,
  h: [33m/pizza/[39m,
  i: [
    [36m1[39m,
    [36m2[39m
  ],
  j: {
    x: [36m1[39m,
    y: [36m2[39m
  }
}"
`;

exports[`stringify() should print contructor names for classes 1`] = `
"{
  x: Pizza {
    name: [32m'Quattro Formaggi'[39m
  }
}"
`;

exports[`stringify() should stringify part of the object if an address is specified 1`] = `
"{
  z: [36m2[39m
}"
`;
